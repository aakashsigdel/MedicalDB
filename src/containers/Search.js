import React, { Component } from 'react'
import {
  Image,
  StyleSheet,
  TextInput,
  View
} from 'react-native'
import Button from '../components/Button'

const SearchBar = () =>
  <View style={styles.searchBarWrapper}>
    <TextInput style={styles.searchBar} underlineColorAndroid={'transparent'} />
    <Image
      source={require('../../images/searchIcon.png')}
      style={styles.searchIcon}
    />
  </View>

export default class SearchPage extends Component {
  render () {
    return (
      <View style={styles.container}>
        <View />
        <SearchBar />
        <Button
          onPress={() => {}}
          backgroundColor={'#A1DE93'}
          color={'#FFF'}
          title={'CATEGORIES'}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between',
  },
  searchBar: {
    height: 48,
    backgroundColor: '#FFF',
    alignItems: 'flex-end',
    justifyContent: 'center',
    paddingRight: 35,
    paddingLeft: 20,
    borderColor: 'rgba(0, 0, 0, 0.1)',
    borderWidth: 1,
    borderRadius: 2
  },
  searchBarWrapper: {
    shadowColor: '#000',
    shadowOpacity: 0.2,
    shadowOffset: {width: 0, height: 0},
    marginLeft: 15,
    marginRight: 15
  },
  searchIcon: {
    position: 'absolute',
    height: 18,
    width: 18,
    right: 20,
    top: 15
  },
})
