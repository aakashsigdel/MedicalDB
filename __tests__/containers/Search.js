import 'react-native'
import React from 'react'
import Search from '../../src/containers/Search'

import renderer from 'react-test-renderer'

it('should render <Search /> correctly', () => {
  const tree = renderer.create(
    <Search />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

