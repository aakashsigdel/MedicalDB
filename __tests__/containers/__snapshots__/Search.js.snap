exports[`test should render <Search /> correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
    }
  }>
  <View />
  <View
    style={
      Object {
        "marginLeft": 15,
        "marginRight": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.2,
      }
    }>
    <TextInput
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#FFF",
          "borderColor": "rgba(0, 0, 0, 0.1)",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 48,
          "justifyContent": "center",
          "paddingLeft": 20,
          "paddingRight": 35,
        }
      }
      underlineColorAndroid="transparent" />
    <Image
      source={1}
      style={
        Object {
          "height": 18,
          "position": "absolute",
          "right": 20,
          "top": 15,
          "width": 18,
        }
      } />
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#0C42FD",
            "borderRadius": 2,
            "elevation": 4,
            "height": 44,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#A1DE93",
          },
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontWeight": "500",
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#FFF",
            },
          ]
        }>
        CATEGORIES
      </Text>
    </View>
  </View>
</View>
`;
